haggis.js

```diff
@@ -1,161 +1,187 @@
 async function parseLogAndPopulateForm() {
-    const stats = parseLog(document.getElementById('logText').value);
-    const data = await serializeJson(stats);
-    document.getElementById('data').value = data;
-    document.getElementById('statsForm').submit();
+  const stats = parseLog(document.getElementById("allLogs").textContent);
+  const data = await serializeJson(stats);
+  document.getElementById("data").value = data;
+  document.getElementById("statsForm").submit();
 }
 
 async function renderStats() {
-    const urlParams = new URLSearchParams(window.location.search);
-    const data = urlParams.get('data');
+  const urlParams = new URLSearchParams(window.location.search);
+  const data = urlParams.get("data");
 
-    if (!data) {
-        document.getElementById("stats").innerHTML = 'Data not found';
-        return;
-    }
+  if (!data) {
+    document.getElementById("stats").innerHTML = "Data not found";
+    return;
+  }
 
-    const stats = await deserializeJson(data);
-    const html = renderStatsAsHtmlString(stats);
-    document.getElementById("stats").innerHTML = html;
+  const stats = await deserializeJson(data);
+  const html = renderStatsAsHtmlString(stats);
+  document.getElementById("stats").innerHTML = html;
 }
 
-function parseLog(text) {
-    const lines = text.split('\n');
-    const game = {
-        players: [],
-        rounds: [],
-        playerStats: {},
-    };
+function parseLog(logLines) {
```

Comment:
My browser is complaining about a balanced `{` issue and I can run the code.

```
Uncaught SyntaxError: missing } after function body[haggis.js:545:1](file:///home/pwinckles/workspace/haggis-stats/haggis.js)note: { opened at line 22, column 29
```

Reply:
fixed